core/test/base/array.cpp

```diff
@@ -304,16 +347,182 @@ TYPED_TEST(Array, ChangesExecutors)
+    EXPECT_EQ(array_size2.get_num_elems(), 2);
+    ASSERT_THROW(view = array_size4, gko::OutOfBoundsError);
+}
+
+
+TYPED_TEST(Array, MoveArrayToArray)
+{
+    gko::Array<TypeParam> array(this->exec, {1, 2, 3});
+    gko::Array<TypeParam> array2(this->exec, {5, 4, 2, 1});
+    auto data2 = array2.get_data();
+
+    array = std::move(array2);
+
+    EXPECT_EQ(array.get_data(), data2);
+    EXPECT_EQ(array.get_data()[0], TypeParam{5});
+    EXPECT_EQ(array.get_data()[1], TypeParam{4});
+    EXPECT_EQ(array.get_data()[2], TypeParam{2});
+    EXPECT_EQ(array.get_data()[3], TypeParam{1});
+    EXPECT_EQ(array.get_num_elems(), 4);
+    EXPECT_EQ(array2.get_data(), nullptr);
+    ASSERT_EQ(array2.get_num_elems(), 0);
+}
+
+
+TYPED_TEST(Array, MoveViewToView)
+{
+    TypeParam data[] = {1, 2, 3, 4};
+    auto view = gko::Array<TypeParam>::view(this->exec, 4, data);
+    TypeParam data2[] = {5, 4, 2};
+    auto view2 = gko::Array<TypeParam>::view(this->exec, 3, data2);
+
+    view = std::move(view2);
+
+    EXPECT_EQ(view.get_data(), data2);
+    EXPECT_EQ(view.get_data()[0], TypeParam{5});
+    EXPECT_EQ(view.get_data()[1], TypeParam{4});
+    EXPECT_EQ(view.get_data()[2], TypeParam{2});
+    EXPECT_EQ(view.get_num_elems(), 3);
+    EXPECT_EQ(view2.get_data(), nullptr);
+    EXPECT_EQ(view2.get_num_elems(), 0);
+    EXPECT_NE(data, nullptr);
+    EXPECT_EQ(data[0], TypeParam{1});
+    EXPECT_EQ(data[1], TypeParam{2});
+    EXPECT_EQ(data[2], TypeParam{3});
+    ASSERT_EQ(data[3], TypeParam{4});
+}
+
+
+TYPED_TEST(Array, MoveViewToArray)
+{
+    TypeParam data[] = {1, 2, 3, 4};
+    gko::Array<TypeParam> array(this->exec, {5, 4, 2});
+    auto view = gko::Array<TypeParam>::view(this->exec, 4, data);
+
+    array = std::move(view);
+
+    EXPECT_EQ(array.get_data(), data);
+    EXPECT_EQ(array.get_data()[0], TypeParam{1});
+    EXPECT_EQ(array.get_data()[1], TypeParam{2});
+    EXPECT_EQ(array.get_data()[2], TypeParam{3});
+    EXPECT_EQ(array.get_data()[3], TypeParam{4});
+    EXPECT_EQ(array.get_num_elems(), 4);
+    EXPECT_EQ(data[0], TypeParam{1});
+    EXPECT_EQ(data[1], TypeParam{2});
+    EXPECT_EQ(data[2], TypeParam{3});
+    EXPECT_EQ(data[3], TypeParam{4});
+    EXPECT_EQ(view.get_data(), nullptr);
+    ASSERT_EQ(view.get_num_elems(), 0);
+}
+
+
+TYPED_TEST(Array, MoveArrayToView)
+{
+    TypeParam data[] = {1, 2, 3};
+    auto view = gko::Array<TypeParam>::view(this->exec, 3, data);
+    gko::Array<TypeParam> array_size2(this->exec, {5, 4});
+    gko::Array<TypeParam> array_size4(this->exec, {5, 4, 2, 1});
+    auto size2_ptr = array_size2.get_data();
+    auto size4_ptr = array_size4.get_data();
+
+    view = std::move(array_size2);
+
+    EXPECT_EQ(view.get_data()[0], TypeParam{5});
+    EXPECT_EQ(view.get_data()[1], TypeParam{4});
+    EXPECT_EQ(view.get_num_elems(), 2);
+    EXPECT_NE(view.get_data(), data);
+    EXPECT_EQ(view.get_data(), size2_ptr);
+    EXPECT_NO_THROW(view = std::move(array_size4));
```

Comment:
When move array to the view, the view become array. Is it correct?

Reply:
Yes.

Reply:
Got it. thanks